airflow/decorators/base.py

```diff
@@ -382,7 +382,15 @@ def expand(self, **map_kwargs: OperatorExpandArgument) -> XComArg:
         prevent_duplicates(self.kwargs, map_kwargs, fail_reason="mapping already partial")
         # Since the input is already checked at parse time, we can set strict
         # to False to skip the checks on execution.
-        return self._expand(DictOfListsExpandInput(map_kwargs), strict=False)
+        if self.is_teardown:
+            if "trigger_rule" in self.kwargs:
+                raise ValueError("Trigger rule not configurable for teardown tasks.")
+            self.kwargs.update(trigger_rule=TriggerRule.ALL_DONE_SETUP_SUCCESS)
+        obj = self._expand(DictOfListsExpandInput(map_kwargs), strict=False)
+        obj.is_setup = self.is_setup
+        obj.is_teardown = self.is_teardown
+        obj.on_failure_fail_dagrun = self.on_failure_fail_dagrun
```

Comment:
These very likely should be done in `_expand` instead to account for other call paths.